### PR TITLE
Outfit rework

### LINKED ISSUE #141: [ FEATURE REQUEST ] - Outfit overhaul/rework
# Overview:
Currently, Natsuki only changes outfits based on the time of day, day of the week and affinity. We'd like to rework the outfit system so that Natsuki can wear customisable and preset outfits on demand when affinity allows for this, as well as to allow Natsuki to forgo things like hairclips/hairbands, wear necklaces/hats properly, etc.

## Must haves:
- The player must be able to ask Natsuki to change outfits to a list of preset outfits (I.E school uniform, casual outfit) at medium affinity. She may refuse to change into things like PJs based on time of day or other factors until later. This may or may not include visual previews.
- The player must be able to create and customise outfit combinations, and ask Natsuki to change to them at high affinity.
- Natsuki must be able to wear outfits consisting of at minimum a set of clothing.
- Natsuki must be able to wear outfits that use all slots (clothing, necklace, hat, etc.).
- Natsuki must be able to wear outfits that use all slot combinations, assuming the minimum requirement of clothing is met.

## Could haves:
- Ideally, the preset and customisable outfits will be able to be previewed either by Natsuki trying them on when selected, or a preview window (thumbnail), or both.
- Ideally, Natsuki will have special dialogue based on having her change into outfits for certain times of day, days of week, etc. For example, she may comment on being asked to change into her school uniform on the weekend, or PJs during the day on a weekday.

## Is this feature request related to a problem? Please describe.
This is a new feature.

### LINKED ISSUE #142: [ FEATURE REQUEST ] - Long hair for Natsuki.
# Overview: so i think that it would be amazing if natsuki have longer hair so yeah!
[brief description of what this aims to achieve]

## Must haves: Long hair 
[features that you deem essential for this issue]

## Could haves: N/A
[features that you deem would be good, but not essential, for this issue]

## Is this feature request related to a problem? Please describe.
[prob a feature ]

### PR DESCRIPTION
**Pull request by Blizzardsev**
This will introduce the following:
- New outfit framework
- Complete overhaul of outfit functionality
- Ability to create, delete, export and load custom outfits
- Ability to pick outfits for Natsuki to wear
- A massive range of new wearables, some unlocked by default

To-do:
- [x] Overhaul outfit framework
- [x] Add outfit unlock topic
- [x] Add custom outfit designer
- [x] Add quit safeguards
- [x] Add time-of-day based changes
- [x] Add remove outfit ability
- [x] Complete testing, general cleanup

closes #141 
closes #142

### CODE REVIEW

**Review by multimokia — CHANGES_REQUESTED** [review]
Some initial stuff

File: game/definitions.rpy [hunk 1]
```diff
@@ -1221,6 +1224,95 @@ init -999 python in jn_utils:
...
+        IN:
+            path - The path to check
+
+        OUT: 
+            - True if the file exists, otherwise False
+        """
+        if os.path.isfile(path):
+            return True
+
+        return False
```
[the remark this answers is not in the document]
Comment (Blizzardsev) [reply game/definitions.rpy:1240]: Good call

File: game/definitions.rpy [hunk 2]
```diff
@@ -1221,6 +1224,95 @@ init -999 python in jn_utils:
...
+            - path - the file path to search
+            - extension_list - optional list of file extensions; only files with these extensions will be returned
+
+        OUT:
+            - Tuple representing (file_name, file_path)
+        """
+        return_file_items = []
+
+        for file in os.listdir(path):
+            if (not extension_list or any(file_extension in file for file_extension in extension_list)):
```
[the remark this answers is not in the document]
Comment (Blizzardsev) [reply game/definitions.rpy:1311]: Resolved

File: game/natsuki.rpy [hunk 3]
```diff
@@ -1,371 +1,489 @@
...
+
+            IN: 
+                - reference_name - The accessory reference name to check if Natsuki is wearing
+
+            OUT:
+                - True if Natsuki is wearing the specified accessory, otherwise False
+            """
+            return Natsuki._outfit.accessory.reference_name == reference_name
+
+        @staticmethod
```
[the remark this answers is not in the document]
Comment (Blizzardsev) [reply game/natsuki.rpy:134]: Gotcha

File: game/outfits.rpy [hunk 4]
```diff
@@ -1,289 +1,2027 @@
...
             self.necklace = necklace
 
-    #TODO: We need a means of storing and recalling user-defined outfit combos, as well as creating them.
-    # For now, we'll settle with presets.
+        @staticmethod
+        def load_all():
+            """
+            Loads all persisted data for each outfit from the persistent.
+            """
+            global __ALL_OUTFITS
```
[the remark this answers is not in the document]
Comment (Blizzardsev) [reply game/outfits.rpy:287]: That's odd; I'd have thought accessing and assigning to the collection would require it, but fair enough

**Pull request merged**